packages/backend-for-frontend/test/getCatalogEServiceDescriptor.test.ts

```diff
@@ -0,0 +1,328 @@
+            explicitAttributeVerification: false,
+            id: certifiedAttributeId,
+            name: certifiedAttributeName,
+          },
+        ],
+      ],
+      declared: [
+        [
+          {
+            description: attributeDescription,
+            explicitAttributeVerification: false,
+            id: declaredAttributeId,
+            name: declaredAttributeName,
+          },
+        ],
+      ],
+      verified: [
+        [
+          {
+            description: attributeDescription,
+            explicitAttributeVerification: true,
+            id: verifiedAttributeId,
+            name: verifiedAttributeName,
+          },
+        ],
+      ],
+    },
+    publishedAt: eServiceDescriptor.publishedAt,
+    suspendedAt: eServiceDescriptor.suspendedAt,
+    deprecatedAt: eServiceDescriptor.deprecatedAt,
+    archivedAt: eServiceDescriptor.archivedAt,
+    interface: undefined,
+    docs: [],
+    eservice: catalogDescriptorEService,
+  };
+
+  const mockCatalogProcessClient = {
+    getEServiceById: vi.fn().mockResolvedValue(eService),
+  } as unknown as CatalogProcessClient;
+
+  const mockTenantProcessClient = {
+    tenant: {
+      getTenant: vi.fn().mockResolvedValue({
+        id: tenantId,
+        name: tenantName,
+        attributes: [],
+      }),
+    },
+  } as unknown as TenantProcessClient;
+
+  const mockAgreementProcessClient = {
+    getAgreement: vi.fn(),
+    getAgreements: vi.fn(),
+  } as unknown as AgreementProcessClient;
+
+  const mockAttributeProcessClient = {
+    getBulkedAttributes: vi.fn().mockResolvedValue({
+      results: [],
+      totalCount: 0,
+    }),
+  } as unknown as AttributeProcessClient;
+
+  const mockDelegationProcessClient = {
+    delegation: {
+      getDelegations: vi.fn().mockResolvedValue([]),
+    },
+  } as unknown as DelegationProcessClient;
+
+  const mockEServiceTemplateProcessClient =
+    {} as unknown as EServiceTemplateProcessClient;
+
+  vi.spyOn(attributeService, "getAllBulkAttributes").mockResolvedValue([
+    {
+      id: certifiedAttributeId,
+      name: certifiedAttributeName,
+      description: "mockDescription",
+      kind: "VERIFIED",
+      creationTime: new Date().toTimeString(),
+    },
+    {
+      id: declaredAttributeId,
+      name: declaredAttributeName,
+      description: attributeDescription,
+      kind: "VERIFIED",
+      creationTime: new Date().toTimeString(),
+    },
+    {
+      id: verifiedAttributeId,
+      name: verifiedAttributeName,
+      description: attributeDescription,
+      kind: "VERIFIED",
+      creationTime: new Date().toTimeString(),
+    },
+  ]);
+
+  vi.spyOn(delegationService, "getAllDelegations").mockResolvedValue([]);
+
+  vi.spyOn(agreementService, "getLatestAgreement").mockResolvedValue({
+    id: generateId(),
+    eserviceId: eServiceId,
+    descriptorId: mockDescriptorId,
+    consumerId: authData.organizationId,
+  } as unknown as agreementApi.Agreement);
```

Comment:
here it might be better to remove the casts, adding the other fields needed, to be closer to the real logic (e.g. spot code that returns too many fields)